### ISSUE TITLE
IPython 5.3.0 debug mode breakpoint issue

### ISSUE THREAD

**Issue by 40f2e71e-2cf0-4fc5-bade-e4b0158190e0** [python/cpython#74555]
BPO | [30370](https://bugs.python.org/issue30370)
--- | :---
Nosy | @berkerpeksag
Files | <li>[ipython_error](https://bugs.python.org/file46864/ipython_error "Uploaded as text/plain at 2017-05-15.14:43:42 by Taylor Newton"): ipython error example</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-05-15.16:09:45.146>
created_at = <Date 2017-05-15.14:43:43.931>
labels = ['type-crash']
title = 'IPython 5.3.0 debug mode breakpoint issue'
updated_at = <Date 2017-05-15.16:09:45.132>
user = 'https://bugs.python.org/TaylorNewton'
```

bugs.python.org fields:
```python
activity = <Date 2017-05-15.16:09:45.132>
actor = 'berker.peksag'
assignee = 'none'
closed = True
closed_date = <Date 2017-05-15.16:09:45.146>
closer = 'berker.peksag'
components = []
creation = <Date 2017-05-15.14:43:43.931>
creator = 'Taylor Newton'
dependencies = []
files = ['46864']
hgrepos = []
issue_num = 30370
keywords = []
message_count = 2.0
messages = ['293712', '293719']
nosy_count = 2.0
nosy_names = ['berker.peksag', 'Taylor Newton']
pr_nums = []
priority = 'normal'
resolution = 'third party'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'crash'
url = 'https://bugs.python.org/issue30370'
versions = ['Python 2.7']
```

</p></details>

**Comment by 40f2e71e-2cf0-4fc5-bade-e4b0158190e0** [comment]
Hi Python Team,

I recently upgraded Python to version 2.7.8 (from 2.6.6) and IPython to version 5.3.0.  I noticed that when using the magic command (%run -d) in IPython, subsequent runs of a program seem to fail with a "KeyError", apparently due to the debugger's failure to clear breakpoints from previous runs of the program.

I attached a sample of output from one such failed run, where a breakpoint that had been set at line 56 in a previous call caused the program to crash, though in the current call, only one breakpoint at line 70 is requested.  

For the time being, I've downgraded to IPython version 4.2.0, which seems to work as expected.

Cheers,
Taylor
